core/src/smartcontracts/isi/triggers.rs

```diff
@@ -174,18 +172,16 @@ pub mod isi {
                             false
                         };
                     if allow_execute {
-                        acc.lock().unwrap().push((id.clone(), authority.clone()));
                         Ok(())
                     } else {
                         Err(ValidationError::new("Unauthorized trigger execution").into())
                     }
                 })
-                .ok_or_else(|| Error::Find(Box::new(FindError::Trigger(id))))?;
-            let new_array = acc.lock().unwrap().clone();
-            for (id, authority) in new_array {
+                .ok_or_else(|| Error::Find(Box::new(FindError::Trigger(id.clone()))))?;
+            if temporary_error.is_ok() {
                 wsv.execute_trigger(id, authority);
```

Comment:
Maybe instead of `if` chain with `.and_then` this way you don't need bind variable at all. 

```suggestion
                .and_then(|_| wsv.execute_trigger(id, authority))
```

Reply:
good suggestion